### PR TITLE
git-vendor: new formula

### PR DESCRIPTION
**Pull request by liamstask**

### CODE REVIEW

File: Library/Formula/git-vendor.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class GitVendor < Formula
+  desc "command for managing git vendored dependencies"
+  homepage "https://brettlangdon.github.io/git-vendor"
+  head "https://github.com/brettlangdon/git-vendor.git"
+
+  stable do
```
Comment (DomT4) [inline Library/Formula/git-vendor.rb]: Don't need a `stable do` block unless it requires something special/unique to stable, whether that be resources, patching, a dependency `head` doesn't, etc. If none of that is the case here, kill the block and wrap it into the initial block:

```ruby
desc ""
homepage ""
url ""
sha256 ""
head ""
```
etc.

Comment (liamstask) [inline Library/Formula/git-vendor.rb]: @DomT4 thanks, pushed an updated version that incorporates that info into the initial block

File: Library/Formula/git-vendor.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+class GitVendor < Formula
+  desc "command for managing git vendored dependencies"
+  homepage "https://brettlangdon.github.io/git-vendor"
+  url "https://github.com/brettlangdon/git-vendor/archive/v1.0.0.tar.gz"
```
Comment (brettlangdon) [inline Library/Formula/git-vendor.rb]: I actually just pushed an updated version, so if we could make this `v1.1.0` that would be greatly appreciated :)

Comment (liamstask) [inline Library/Formula/git-vendor.rb]: updated to `v1.1.0`

**Comment by liamstask** [comment]
@DomT4 just wanted to clarify this is ready to go, no further edits planned.

File: Library/Formula/git-vendor.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  url "https://github.com/brettlangdon/git-vendor/archive/v1.1.0.tar.gz"
+  sha256 "1a4db75631af559152f19b41867c23c0fe6c4ff30117afa6ab2d02a5802b89e6"
+  head "https://github.com/brettlangdon/git-vendor.git"
+
+  def install
+    system "make", "PREFIX=#{prefix}", "install"
+  end
+
+  test do
+    system "git", "vendor", "--help"
```
Comment (mikemcquaid) [inline Library/Formula/git-vendor.rb]: Would be good to see this test actually add a vendored dependency.

Comment (liamstask) [inline Library/Formula/git-vendor.rb]: @mikemcquaid done

File: Library/Formula/git-vendor.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    system "git", "init"
+    system "git", "config", "user.email", "author@example.com"
+    system "git", "config", "user.name", "Au Thor"
+    system "git", "add", "."
+    system "git", "commit", "-m", "Initial commit"
+    system "git", "vendor", "add", "forge", "https://github.com/brettlangdon/forge", "v0.1.4"
+  end
```
Comment (apjanke) [inline Library/Formula/git-vendor.rb]: How about an `assert_match "forge@v0.1.4", shell_output("git vendor list")` to test reading the vendor list back out?

Comment (mikemcquaid) [inline Library/Formula/git-vendor.rb]: :+1:

File: Library/Formula/git-vendor.rb (hunk 3 above)
Comment (mikemcquaid) [inline Library/Formula/git-vendor.rb]: Can remove this line now that you have a great test below :+1:

File: Library/Formula/git-vendor.rb (hunk 4 above)
Comment (brettlangdon) [inline Library/Formula/git-vendor.rb]: My suggestion is to have this test vendor `git-vendor` itself, that way this assertion doesn't rely on an additional repo + tag existing (I don't have any plans to get rid of `forge@v0.1.4`, but better safe than sorry).

**Comment by liamstask** [comment]
@mikemcquaid @apjanke @brettlangdon done, done, and done :D

**Comment by apjanke** [comment]
...and merged.

Thank you for your contribution to Homebrew!

**Comment by brettlangdon** [comment]
@liamstask thank you for taking this on. I appreciate it!

**Comment by liamstask** [comment]
np, thanks for getting it in

**Pull request closed without merging**